### PR TITLE
Improve coverage tool for P3 Analysis Library integration

### PR DESCRIPTION
**Pull request by Pennycook**
This PR introduces a new standalone `cbicov` tool that reads a compilation database and produces JSON that can be read in by the P3 Analysis Library.  It is intended to be invoked as below:
```sh
cbicov compute -S <source directory> -o <output file> compile_commands.json
```

### CODE REVIEW

File: MANIFEST.in
Comment (rfbgo) [inline MANIFEST.in:4]: I think you forgot to add this file to your PR :)

**Review by rfbgo — CHANGES_REQUESTED** [review]

File: MANIFEST.in
Comment (Pennycook) [reply MANIFEST.in:4]: How embarrassing!  Fixed in 57f4ff8449d0874e57fbe30e750d73146df4f4fc.

**Comment by Pennycook** [comment]
@laserkelvin, @rfbgo, @douglasjacobsen, @kaanolgu, @tomdeakin, @swright87 - Please take a look at this.

Things had gotten too far behind main to be merged as-is, and rebasing was giving me a headache, which is why I had to force-push.  I took the opportunity to clean a few things up and make some improvements based on our experience with the P3 automation framework and anticipated future features.  The individual commit messages should explain what's going on, but to summarize:

- `cbicov` can now be invoked via either `cbicov` or `python3 -m codebasin.coverage` thanks to the magic of pyproject.toml.
- `cbicov -S <source-dir> <database.json> <coverage.json>` is now `cbicov compute -S <source-dir> -o <coverage.json> database.json`.  The intent here is to give us flexibility to extend the workflow with commands like `cbicov convert` (for compatibility with other coverage formats), `cbicov show` or `cbicov report` (to help humans interact with the coverage file), etc.
- I trid to split things out such that there is a `cli` function accepting an `argv`, so that we can write tests.

Some specific requests:
- @rfbgo: Please make sure that your srcipts still work if you convert the command style.
- @laserkelvin: Please review the design and let me know if there's anything that looks weird to you.
- @swright87: Please try and run this on a few sample codebases and see if the features you requested look good.

**Review by laserkelvin — APPROVED** [review]

**Comment by Pennycook** [comment]
I spoke to @rfbgo offline. I've decided to go ahead and merge this in as-is, based on the following:
- We know that the earlier design is compatible with an automated workflow, and the fundamentals haven't changed.
- The new design has approval from @laserkelvin.
- We're working on a lot of other functionality concurrently, and we don't want to risk this PR getting stale (again).

If anybody finds any bugs and/or has any feature requests once this new functionality is merged, we can address it through additional issues and PRs.

**Comment by swright87** [comment]
Apologies for not looking at this :) Will poke it in the new year, anyway!

**Pull request merged**
